src/main/scala/it/pagopa/interop/apigateway/service/impl/TenantProcessServiceImpl.scala

```diff
@@ -37,11 +38,28 @@ class TenantProcessServiceImpl(invoker: TenantProcessInvoker, api: TenantApi)(im
     result <- invoker.invoke(request, "Invoking getTenant", handleCommonErrors(s"getTenant ${id.toString}"))
   } yield result
 
+  override def revokeAttribute(origin: String, externalId: String, code: String)(implicit
+    contexts: Seq[(String, String)]
+  ): Future[Unit] = for {
+    (bearerToken, correlationId, ip) <- extractHeaders(contexts).toFuture
+    request = api.m2mRevokeAttribute(
+      xCorrelationId = correlationId,
+      origin = origin,
+      externalId = externalId,
+      code = code,
+      xForwardedFor = ip
+    )(BearerToken(bearerToken))
+    () <- invoker.invoke(request, "Invoking revokeAttribute", handleCommonErrors(code))
+  } yield ()
+
   private[service] def handleCommonErrors[T](
     resource: String
   ): (ContextFieldsToLog, LoggerTakingImplicit[ContextFieldsToLog], String) => PartialFunction[Throwable, Future[T]] = {
     (contexts, logger, msg) =>
       {
+        case ex @ ApiError(code, message, _, _, _) if code == 400 =>
+          logger.error(s"$msg. code > $code - message > $message - ${ex.getMessage}")(contexts)
+          Future.failed(GatewayErrors.UnexistingAttribute(resource))
```

Comment:
I suggest to rename the error to something like
```suggestion
          Future.failed(GatewayErrors.TenantProcessBadRequest(resource))
```
because a 400 on this call may (now or in the future) be related to other reasons

Reply:
Does that exception exist?

Reply:
no no, it was just a renaming idea

Reply:
Is it better now?

Reply:
❤️